docker-compose.yml

```diff
@@ -0,0 +1,47 @@
+# docker-compose setup for a development environment.
+# To use this file, populate config-docker.env with the following:
+#
+# MYSQL_PASSWORD - a password for the mysql user
+# MYSQL_ROOT_PASSWORD - a password for the mysql *root* user
+#
+# Then you can run:
+#   - docker-compose build
+#   - docker-compose up
+
+version: '3.5'
+
+services:
+  app:
+    build: ./
+    restart: always
+    ports:
+      - "8000:3031"
+    env_file:
+      config-docker.env
+    environment:
+      DB_HOST: db
+      DB_PORT: 3306
+      APP_DB_NAME: agagd
+      AGAGD_USER: agagd
+      SECRET_KEY: insecure-key-for-testing
+      DEBUG: "true"
+      LOAD_FIXTURES: "true"
```

Comment:
I realised this after running things locally with no dependencies i.e., db and connecting to preexisting local database. The docker-compose file environmental variables take precedence to the `env_file`, so I cannot connect to my preexisting local database. Therefore, these set of environmental defaults may be better placed in the config-docker.env.sample.

https://docs.docker.com/compose/environment-variables/#the-env-file
>  When you set the same environment variable in multiple files, here’s the priority used by Compose to choose which value to use:
> 
> 1. Compose file
> 2. Shell environment variables
> 3. Environment file
> 4. Dockerfile
> 5. Variable is not defined

Reply:
Hm, I'd personally much rather have the docker-compose approach default to operating as a single cohesive unit as configured, as it seems to me that's the point behind docker-compose. (and I'd prefer to avoid non-'secret' values in a separate env file, or really any mutation to config that's out of band like sample files, as that way lies maintenance debt, e.g. the required variables in `docker-config.env` change, which breaks your setup on your next `git pull`, which leads to digging through changes until you realize the sample file has changed)

In general, my recommendation for users who really want to connect to a non-docker-managed DB and still use docker-compose (for the easy environment variable / docker command-line management, I suppose) would be to put the appropriate configuration into `docker-compose.override.yml`. I can add documentation to that effect if desired.

Reply:
Your suggestion of adding docker-compose.override.yml to the documentation would be appropriate and sounds good. There may be the odd case where it is appropriate to use an external non-docker-managed DB.

Reply:
Resolved unless this issue comes up again.